### PR TITLE
[ALLUXIO-2119] Change SpaceReserver test to not use Whitebox

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-2119

### CODE REVIEW

File: core/server/src/main/java/alluxio/worker/block/SpaceReserver.java [hunk 1]
```diff
@@ -71,33 +65,6 @@ public SpaceReserver(BlockWorker blockWorker) {
...
-  public void run() {
-    long lastCheckMs = System.currentTimeMillis();
-    while (mRunning) {
-      // Check the time since last check, and wait until it is within check interval
-      long lastIntervalMs = System.currentTimeMillis() - lastCheckMs;
-      long toSleepMs = mCheckIntervalMs - lastIntervalMs;
-      if (toSleepMs > 0) {
-        CommonUtils.sleepMs(LOG, toSleepMs);
-      } else {
-        LOG.warn("Space reserver took: {}, expected: {}", lastIntervalMs, mCheckIntervalMs);
```
Comment (gpang) [inline core/server/src/main/java/alluxio/worker/block/SpaceReserver.java]: Does `HeartbeatExecutor` print out messages like this? Would this message be useful?

File: core/server/src/main/java/alluxio/worker/block/BlockWorker.java [hunk 2]
```diff
@@ -199,8 +198,9 @@ public void start() throws IOException {
     getExecutorService().submit(mSessionCleaner);
 
     // Start the space reserver
-    if (mSpaceReserver != null) {
-      getExecutorService().submit(mSpaceReserver);
+    if (spaceReserver != null) {
```
Comment (gpang) [inline core/server/src/main/java/alluxio/worker/block/BlockWorker.java]: Is `spaceReserver` allowed to be created here? Then, we could just check for the configuration here, create the `SpaceReserver`, and directly add it to the threadpool.

**Comment by gpang** [comment]
@aaudiber thanks for this improvement! I left a few minor comments.

File: core/server/src/main/java/alluxio/worker/block/SpaceReserver.java (hunk 1 above)
Comment (aaudiber) [inline core/server/src/main/java/alluxio/worker/block/SpaceReserver.java]: The `SleepingTimer` we switch to has the same functionality:
```java
      long executionTimeMs = System.currentTimeMillis() - mPreviousTickMs;
      if (executionTimeMs > mIntervalMs) {
        LOG.warn("{} last execution took {} ms. Longer than the interval {}",
                mThreadName, executionTimeMs, mIntervalMs);
      } else {
        Thread.sleep(mIntervalMs - executionTimeMs);
      }
```

File: core/server/src/main/java/alluxio/worker/block/BlockWorker.java (hunk 2 above)
Comment (aaudiber) [inline core/server/src/main/java/alluxio/worker/block/BlockWorker.java]: Done

**Comment by gpang** [comment]
LGTM

**Pull request merged**